.github/workflows/release.yml

```diff
@@ -0,0 +1,280 @@
+             bash check_7za.sh `pwd`/../bin/7za
+             bash check_7zr.sh `pwd`/../bin/7zr
+             cd ..
+             ./bin/7za a macos-p7zip.7z ./bin/* -t7z -y
+             ./bin/7z a macos-p7zip.zip ./bin/* -tzip -y
+      - name: Upload the macOS binary artifacts
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: macos-p7zip.*
+          asset_name: macos-p7zip
+          tag: ${{ github.ref }}
+          overwrite: true
+          file_glob: true
+          body: "This is macos binary release using make"
+
+  ubuntu-cmake:
+    name: ubuntu-cmake-binary
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: cmake build 7z_ & test
+        run: |
+             cd ./CPP/7zip/CMAKE/
+             rm -rf ./build
+             mkdir build
+             cd build
+             cmake ..
+             make 7z_
+             make 7z
+             make Rar
+             make 7zCon.sfx
+             cd ./bin
+             ./7z_ a t_.7z ../../../Archive
+             ./7z_ x t_.7z
+             echo "compare file"
+             diff -r ../../../Archive ./Archive
+             echo "delte tmp file"
+             rm -rf t_.7z Archive
+
+      - name: cmake build 7za & test
+        run: |
+             cd ./CPP/7zip/CMAKE/build/
+             make 7za
+             cd ./bin
+             ./7za a ta.7z ../../../Archive
+             ./7za x ta.7z
+             echo "compare file"
+             diff -r ../../../Archive ./Archive
+             echo "delete tmp file"
+             rm -rf ta.7z Archive
+
+      - name: cmake build 7zr & test
+        run: |
+             cd ./CPP/7zip/CMAKE/build/
+             make 7zr
+             cd ./bin
+             ./7zr a tr.7z ../../../Archive
+             ./7zr x tr.7z
+             echo "compare file"
+             diff -r ../../../Archive ./Archive
+             echo "delete tmp file"
+             rm -rf tr.7z Archive
+
+      - name: check
+        run: |
+             cd ./check
+             bash ./check.sh `pwd`/../CPP/7zip/CMAKE/build/bin/7z_
+             bash ./check_7za.sh `pwd`/../CPP/7zip/CMAKE/build/bin/7za
+             bash ./check_7zr.sh `pwd`/../CPP/7zip/CMAKE/build/bin/7zr
+             cd ..
+             rm -rf ./CPP/7zip/CMAKE/build/bin/Codecs/CMakeFiles
+             rm -rf ./CPP/7zip/CMAKE/build/bin/Codecs/Makefile
+             rm -rf ./CPP/7zip/CMAKE/build/bin/Codecs/cmake_install.cmake
+             ./CPP/7zip/CMAKE/build/bin/7za a linux-cmake-p7zip.7z ./CPP/7zip/CMAKE/build/bin/7z* ./CPP/7zip/CMAKE/build/bin/Codecs -t7z -y
+             ./CPP/7zip/CMAKE/build/bin/7z_ a linux-cmake-p7zip.zip ./CPP/7zip/CMAKE/build/bin/7z* ./CPP/7zip/CMAKE/build/bin/Codecs -tzip -y
+      - name: Upload the cmake Linux binary artifacts
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Can you explain the work flow here? I'm not very good at git action. 
Should I set a secrets.GITHUB_TOKEN ?
Thanks!

Reply:
No, that is provided, pulled in by GitHub Actions. The action and binaries for release are only produced when you go and actually create an release tag.

As i have here: https://github.com/techno-express/p7zip/releases 

This is from https://github.com/marketplace/actions/upload-files-to-a-github-release